### ISSUE TITLE
Filter causes YAMLException

### ISSUE THREAD

**Issue by eurog33k** [apideck-libraries/portman#631]
I'm issuing the following command:
portman -l ./time.json -o ./outcoll.json --filterFile ./filter.json

My portman version is 1.29.1 on MacOS 13.6.7

I'm getting the following:   
  

> ======================================================================================================================================================================================================================================
 Local Path:  		./time.json
 Output Path:  		./outcoll.json
 Portman Config:  	portman-config.default.json
 Postman Config:  	postman-config.default.json
 Filter Config:  	./filter.json
 Environment:  		.env
 Inject Tests:  	true
 Run Newman:  		false
 Newman Iteration Data: false
 Upload to Postman:  	false  
======================================================================================================================================================================================================================================
err YAMLException {
  name: 'YAMLException',
  reason: 'unacceptable kind of an object to dump [object Undefined]',
  mark: null,
  message: 'JS-YAML: unacceptable kind of an object to dump [object Undefined]',
  isWarning: false
}
 OAS File Error - Unable to process the OpenAPI file.
 Error: Filter file error - ./filter.json doesn't exist. 
 ======================================================================================================================================================================================================================================

I'm attaching my openapi 3 file and my filter file
[time.json](https://github.com/user-attachments/files/16612897/time.json)
[filter.json](https://github.com/user-attachments/files/16612898/filter.json)

I haven't been able to figure out why the filter doesn't work.

If I issue the command without the filter (portman -l ./time.json -o ./outcoll.json), the outcoll.json Postman collection gets generated correctly.

I'm unable to get the filter to work.

**Comment by thim81** [comment]
Hi @eurog33k 

We look into the cause of this, which we can try to reproduce thanks to your clear explanation and files. It could be that the filter action results in a incorrect YAML structure. We will keep you posted of our findings.

**Comment by thim81** [comment]
hi @eurog33k 

The issue lies with the empty oneOf:

<img width="746" alt="image" src="https://github.com/user-attachments/assets/af83fcd2-1224-4a63-a69d-4d5a3e1a7ccf">

The filtering mechanisme cleans up "empty" objects. 
In case of anyOf or oneOf; the empty objects are valid so they should be kept.

This will be fixed in the next release of Portman.

As a work-around, you can already keep the "empty" objects by adding `"preserveEmptyObjects": true` in your filters.

filter.json >>
```json
{
  "methods": ["POST","DELETE"],
  "tags": [],
  "operationIds": [],
  "operations": [],
  "flags": [],
  "flagValues": [],
  "preserveEmptyObjects": true
}
```

You can see the filtered result in the [online playground](https://openapi-format-playground.vercel.app/?input=H4sIAAAAAAAAA%2B1cbW%2FbOBL%2Bnl%2FBwx3gpqjsvu3h4E%2BXbQtssNlr0GQXBQ6HMy3RNjcSqZJSUu%2Fe%2FvcbUpRFSpQs2W7aZO0Puw1fhsN5eTgcD81TwnBKp%2BjV%2BPn41QllCz49QSijWUym6Hue30m%2ByNDZ5fntS9VORCLfL66IuKUhDBitsiydTiYxWeJ4PDfDx3MyGcHokLMMh5kiiBDDCUx4S8WN%2FpMkmMZTBGv%2FM5eEZkvB83Qc8kT35iKuiLv9ExZPDGG1xi0RknI2RS%2FHL8avx9%2F948VJhpdyiv79n5MUZyupVp9kNCGTCFZcJxj2QHEsC66WxLCHEE%2BJwBnQOo%2Bm6DLGjBHxcyqJyMYw6q2a%2FFM5%2BYLKzEzTq5l%2FIxTAjtKYhpqQ1aoYMH%2BmWIAogJAzjcIWPuVErDdtpchSwX8lYWa1C%2FIpp4IAmwvghVg9MlyBYKdWCzC4ToEKy5M5EU7HgguQxhRFPJ%2FHNhXyOY15BJMykTvUs7WyCjWvF%2BckATprQh4g6xFoek%2B2ZSYoW%2B7F24rgyNl5wdxZGJI0Cy4wW%2BZ4aZOKiAwFTTPtEf8LHK7O0B2hy1VGIjApsiCCsJCgGCwZ8QWKDTFtz2N0vSIITB%2B8C93ROIZ%2Fx2CDKIPmOYEZIP1whTKOYh7imP5GdJcgMuVMkrGz8PkCMY5mTz4GpzXOZ2aHEmFBlKHf0gjY40JNwHosBv1WzIU8jyPgwPBDomdqXaYXt9mNyALncaYYzCVoQQ8AHBEjuSH2DNGFgqgFXeZCUYJl37ElCGSFsBoG24QNcfD%2FE2c%2FV3maQisw6ghtimYsnj1DM8LUfxdi9sUMqEGjsCBJweFIPxv6GHz7VoTqZlRaiKX1WYtJoadXP5xdXDxVxkKXjIPsj1b59a2y1K119o1ePn8%2Bslf6G9gVnP5%2FhWM%2BgcGEZXKymTe5hpP0rXOSfzB9729GFdHXNaKuTTv7%2BkCyXDCJiBAg7HIlFOVEaQoY0Hpa8Djmd0prgmAJI1zhPEXn7BZs2BU%2FzMpZBH7YYmhKAcoRGk6qCF5rXygGzHm0RhR4pMCLMsQMYcUPqHueZ5ZjwA6wDrxAbDXyPeaP0ZNCCqHW6IwWewqU6menNf5%2BUV060AHV05hEtf4zBJNBiIALmCFDSlsRWoEpzwm4iExJSBeUaBnpo7iKj9DMhD6zIVztv2oZttzzsirk%2BLJLGjOw1mwY5P0u3ziEejNgENr1QSB3eR78Ym4ETpcfVQyVycezlJppoxrFD4ULBXApGELQTDuPmvQKeAkUjg0kWcxUE22qxtldWtYtZPKrbErDh%2B96Xhy%2FXzSbVTDhY64gIyfvlNJKBkfe6TDDT7noLg4XPq9dddwP4AFc0jJKZBsh9dEW9JZkgEmd48oTDQuB153jaEaSLaRAdGzdvsHqE7TL2De4S%2BrmxLmGXWgF%2BAQ%2FlOYPcOfWxK4FpplsIxmg3%2F%2FYdJm4pv8Z%2FraY4LGalMs%2BF%2FJc%2F8%2B9kx%2F%2BPt4SOr8pHC54xwCl3FjIiTJG75lGRx0GbqKz8mwmZraK0Ja%2F0VRFZyDIWJ0CttS%2FuZtDuf0LwpbZ6itHkxplv4fgqCLuwcNuNPRjYTs2tKBBl1c5lvqh4rpSdCU8KzvwiKPlXYNbmqmRpbR0f0Kluk1Z3WBVBGy16Na3RNdoZ1ZEXJFaw1DFBYtSTlm2byzdze2gSFvfo9X12KBJXXAxkRI6cHHlpCxSxl5uqX33NWouQxnngUxgYw1u3hX7VlPrYFYnCbwp8WzQT8vUnqjDRf8kZUoVbNa10yY%2FwvLky99UmqFs%2FVC471C6ruIvubpjfk9mdXSdnQ5ZXR%2BSAiVcEJSzG8bvGHpSxIGn6IasJTBxSw7AhSEeKJrHu8TxLnG8S3wjd4kvsp9rzq%2FU6fUoNmOU8w7Otkexn58LKP4RkPirXlxHr1%2B8ao97O6MuE%2FyRzyEhkdT9Cf5MkzxBkmTqyzw3XjHRoD1e094xWBs%2FPNDeisrbEbkfGvdB4q0o3MPWtwLQBRbLPmi6s2EfIxfzAUf%2Brt2RR9%2FGjeUBBlp%2FMp8dds4e3XYISZ%2FbHixxPPJVeE1%2Bd%2F7%2BL43%2BMBgRkRhuzT3yzMVA3ValKQ%2BYYVYVao0Ea51rb4q1XkPkgYZOS48uS6ZsJR%2Fgu%2FzX7Tj8vhS0ztHlYUikXOTx%2BOgavhPtwaRk76veYHCeSnmX%2Bw2761iDsmRXG8rY8JGtsD76M86RzpSqNc2Gi3uDMCN34MWffj26x%2BbzQCKoY6qq%2BBxTVd9AqupA8ZYkYS5opr9zDXRFpJTX%2FIYwXXWvqIYxBUr%2FEqbBGXQGDNTHlW0VE4q22U%2FBbttXqOVmPJ7k8xvqAFpLgXlLebkpRzvfg0RZW7YPDVUo1pxd%2ByZd8lyEnmF9Fynm78MmTnju4vOw%2BRAzbN%2BmG8lXw%2BacxwQzq720fSe03cTTVly%2B0bLVVqnNanTeCARGYo0GZ0ohE589F4NLH7tHo94UcfeQdRWBbjW%2FXJ3ERAAEdA%2F1aYDaItvwZ0u%2B4sQry%2FuW4REYHhswaJ1Gebj1rGyp8HHduTqBExyuKCMHpfoVQcwJDAY6nMxwlss36rrT0BKF2H7ptQXoefWy8hq1%2FlZbSCD0wMttNuMTY8WjLUe1qPW3Ia9baqHXUAzyAGvjbtB2G9hcLv2VbI1KOy0%2F8LlmCBvo67%2Bn2fNMoeio5bHK5vrbOpPh8vI3oNKuZUNNBZbMGcF0HTkFIlhpMVd1rkH6jDGhTH37N0XPTQP5HMa5pLfkp7Jnk7Or55kHmkno9ZlmKWVDuQGyy6ROnPm70bvdVFZVyMsT7DPkXvSWMZ%2FjeKDn6lVLFL%2FioL6VSnPdqZTUneBs%2BZedYhiL6FmVb%2BJiU6akNmAyPM0EVJXVBBAJb7auplyoOajuFR6P8GYYglYgLT5%2BOHXn1hjkWmry8Dx63vfaLDbe%2BHbwKEhIwOMcCPCs3TJ7K1NelnxwEronhiJpn7naQ%2BzT2obPoJS01VLuq8SJ4nl%2B%2BeJM4UhxFJt%2BJ1e8h%2B3iKKKKCI4vKxiy4qR6Jub%2BccyR7BHHtuLYkBT6o0Mw8jnVz3N3R4f2iMmfTYVIii4pa6Z2A89FpOyp%2FfZA2byIOfYRahNQgEYsj%2BNmQjGAKMy3bgsOB23p6cAvIdWRMzA5uOdE3j6NQD6RrBsbbxRklTp06jC0UXt1uH9UWXz8am9XfIfqu5Tfqv52A%2BgygQ4jaDODDkPoMIUOY%2Bg2hy6D2Mkk5Hab8Au%2FnawZj2IwHGGBuk2zf7iwfT2gBff9sHM9r7DbSRbDS4oqDiB6FffE2z96OuKj0%2FEV8LHU4RZ8hJs47F39jEWjlKSWYK2RLU3IJnn4CLg0XH%2B8q%2BLNjnh3lxhnW6T7o1KJXnJzcgz8Ss%2BpB72%2FQFnVEcRq3WOU3J9oZ6mHlmZbqceji5kpM1ms7al0U%2F%2Fe8%2FTw89jOYe95f3%2B96dMqud7d3puHlvfA8hxJ3tMiUE8D6i2%2BFIQF9IfH1o1GrTajOqtlIzoXggHFkIaxdvw9gPdsQ2MHXu8XS3VN1hFL%2BxPdvWzuT42l5juDI5Y%2BPCwtVNcPSzWM7YqlvbxnG5bW3%2BLdH5zaD7KPiNqb6E4P5h8dlqqN7sRgz2%2F%2B98MFy6SNsxs3Q8rP2v19L912e7pTOm5Vfk99fJQvCnTyoBhX5hLOLs8LEFphFsUq0dB8lep9plKvvfYowvv8pPn0pCoz38p8zugnVYAPAColD6l%2BaXsHcOmwjS7h8JWkgPOIFO%2F27tRPeAqiXvMpD1a5BCkrXO21x4FPcPrvv6qJ3yoC9RAJPYFTIqFxTCUJOYvkaaHDUqGMEPWLNRk3Wr13pdbeEvnqjr0brRCWAHELZs3gxiuBrhcCQx9Recr%2Fu35vqVNT5S%2Fo6nchKKn9qNne22j7NsT7XciQijSnEq3xPKX1acrgZyltT1KGPkfpeIoy6BlKWQN%2FpeRSmq1dCu%2BEUTilAPymqfHbatWvepvJdWJnVc1%2Bf3qF7AuqQUV2U5o%2FmOAbM9MhswdrBb0ACP4fH6ItKIZgAAA%3D&config=H4sIAAAAAAAAA2WQzQrCMBCEX6XkXF%2BgV40gFCqkeLEeVrtNI2kSko1QxHc3%2FaEHvc3sDMPHvlmwnljRgQ6Ys05pQi8wXdiA1Ns2FI3Jsl12rkS9qAMvec0bQyBDkV1vjbEOPZCy5tT%2BXlbf6a08yQvoiKt3HgP6F%2FLB0Vjdn%2FiglJCP2BiWz3gLTzIqHGfAys3be6s1uIDtxq%2BCSP3%2FeJrLmY3kIpVgZASJaXKEQadZB9SL%2BQ3MeiWVAc0%2BX97WKX8aAQAA)

**Comment by thim81** [comment]
FYI: Thanks to your example we also spotted a [unwanted removal of the "delete" property](https://github.com/thim81/openapi-format/pull/125), which will also be fixed in the next Portman version.

**Comment by thim81** [comment]
hi @eurog33k 

We just released [Portman 1.29.2](https://github.com/apideck-libraries/portman/releases/tag/v1.29.2), which contains the fixes for the reported YAML error. 

Thanks for reporting the issue and helping us to improve Portman.

**Comment by eurog33k** [comment]
I can confirm that the update fixes the problem.
I'm glad my report helped the project.
